lib/net.c

```diff
@@ -336,16 +339,26 @@ init_port(struct net_config *net_conf, uint8_t port_id, uint8_t *num_succ_ports)
 	/*
 	 * The following code ensures that the device is ready for
 	 * full speed RX/TX.
+	 *
 	 * When the initialization is done without this,
 	 * the initial packet transmission may be blocked.
+	 *
+	 * Optionally, we can wait for the link to come up before
+	 * continuing. This is useful for bonded ports where the
+	 * slaves must be activated after starting the bonded
+	 * device in order for the link to come up. The slaves
+	 * are activated on a timer, so this can take some time.
 	 */
-	rte_eth_link_get(port_id, &link);
-	if (!link.link_status) {
-		RTE_LOG(ERR, PORT, "Querying port %hhu, and link is down!\n",
-			port_id);
-		ret = -1;
-		return ret;
-	}
+	do {
+		rte_eth_link_get(port_id, &link);
+		if (!wait_for_link && !link.link_status) {
+			RTE_LOG(ERR, PORT,
+				"Querying port %hhu, and link is down!\n",
+				port_id);
+			ret = -1;
+			return ret;
+		}
```

Comment:
Shouldn't one wait 0.5s if we're going to loop? What about a maximum number of trials?

Without the wait, I understand that the log entry will be issued quickly until the port is up. And if the port never comes back, the log entry will never stop.